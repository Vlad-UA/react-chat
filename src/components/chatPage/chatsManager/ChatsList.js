import React from 'react';

import List from 'material-ui/List';

import ChatItem from './ChatItem';

const ChatsList = ({classAdditional, chatsList}) => {
  return (
    <List className={classAdditional}>
      {chatsList && chatsList.map((chat, index) => <ChatItem key={index} {...chat}/>)}
    </List>
  );
}

export default ChatsList;
