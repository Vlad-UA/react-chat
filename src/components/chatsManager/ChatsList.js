import React from 'react';

import List from 'material-ui/List';

import ChatItem from './ChatItem';

class ChatsList extends React.Component {
  render() {
    const {classAdditional, chatsList} = this.props;

    return (
      <List className={classAdditional}>
        {chatsList.map((chat, index) => <ChatItem key={index} title={chat.title}/>)}
      </List>
    );
  }
}

export default ChatsList;
