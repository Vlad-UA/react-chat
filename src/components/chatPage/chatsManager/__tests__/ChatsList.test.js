/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatsList from '../ChatsList';

jest.mock('../ChatItem', () => () => 'ChatItem');

const mockProps = {
  chats: [{
    _id: '5b07ff83b425b605442572c4',
    updatedAt: '2018-05-30T11:15:38.127Z',
    createdAt: '2018-05-25T12:20:19.920Z',
    creator: {
      _id: '5b032bc2b425b605442572c0',
      username: 'username',
      lastName: 'Baz',
      firstName: 'Denys',
    },
    title: 'React',
    members: [
      {
        _id: '5b097a59b425b605442572cc',
        username: '1',
        lastName: '1',
        firstName: '1',
      },
    ],
    __v: 0,
  },
  ],
  disabled: false,
};

describe('<ChatsList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatsList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot renders correctly - renders correctly with chats', () => {
    const tree = renderer.create(<ChatsList {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - renders without chats', () => {
    const tree = renderer.create(<ChatsList {...mockProps} chats={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
