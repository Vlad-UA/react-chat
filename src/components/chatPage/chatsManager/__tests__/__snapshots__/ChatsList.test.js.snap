// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatsList /> snapshot renders correctly - renders correctly with chats 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatsList-root-7"
>
  ChatItem
</ul>
`;

exports[`<ChatsList /> snapshot renders correctly - renders without chats 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatsList-root-7"
>
  <h3
    className="MuiTypography-root-13 MuiTypography-subheading-20 ChatsList-noChatMessage-8"
  >
    There is no chats yet...
  </h3>
</ul>
`;
